code/AssetLib/Assbin/AssbinLoader.cpp

```diff
@@ -91,7 +91,12 @@ bool AssbinImporter::CanRead(const std::string &pFile, IOSystem *pIOHandler, boo
     }
 
     char s[32];
-    in->Read(s, sizeof(char), 32);
+    const size_t read = in->Read(s, sizeof(char), 32);
+
+    pIOHandler->Close(in);
+    if (read < 19) {
+      return false;
+    }
 
     pIOHandler->Close(in);
 
```

Comment:
How about doing this instead?
```
    char s[32];
    const size_t read = in->Read(s, sizeof(char), 32);

    pIOHandler->Close(in);

    const char *id = "ASSIMP.binary-dump.";
    if (read < strlen(id)) {
      return false;
    }
    return strncmp(s, id, strlen(id)) == 0;
```

Reply:
I am not sure about the string style of Assimp so I would keep the way `strncmp(s, "ASSIMP.binary-dump.", 19) == 0` as is instead of creating a variable like `const char *id = "ASSIMP.binary-dump.";`

Reply:
I guess the best way is just using a scoped pointer for the filestream and let it get closed automatically. I will solve this afterward when this one is merged.